stack.py

```diff
@@ -57,22 +57,77 @@ def parse_program(code):
     [Instr('push', [1.0])]
     >>> list(parse_program('↔'))
     [Instr('swap')]
+
+    Prefixes are placed before the instruction
+    Some prefixes also have sigils
+    >>> list(parse_program('♯ +'))
+    [Instr('add', [], ['quiet'])]
     """
-    for line in re.split('\n|;', code):
+    split_program = re.split('\n|;', code)
+    label_indexes = get_label_indexes(split_program);
+    # Represents ONLY instruction indexes
+    # Used to map labels and index numbers.
+    current_index = 0
+
+    # TODO: Refactor into own function.
+
+    current_index = 0
+    for line in split_program:
+        # TODO: Disallow `@label add 1`, `add @label 1`, etc
+        # `@label; add 1` is ok though.
         parts = line.strip().split()
-        if not parts:
+        # Ignore newlines 
+        if not parts or is_label(parts[0]):
             continue
-        # TODO: This only checks for 'quiet' as a prefix but should allow
-        # "#" and "♯" as equlivlent (move it to Instr.)
-        if parts[0] == 'quiet':
-            prefix = ['quiet']
-            op = parts[1]
-            args = [float(arg) for arg in parts[2:]]
+        # Process an actual instruction
         else:
-            prefix = None
-            op = parts[0]
-            args = [float(arg) for arg in parts[1:]]
-        yield Instr(op, args, prefix)
+            prefix = []
+            args = []
+            for part in parts:
+                if part in prefixes:
+                    prefix.append(prefixes[part])
+                # We convert labels to their relative jump form
+                # TODO: This also converts labels on their own line
+                # Ex: @label_here == 2
+                # Which might be bad?
+                elif is_label(part):
+                    op = 'to'
+                    # Increment the index by one because we
+                    # jump to the instruction right after the label
+                    args.append(float(label_indexes[part]) + 1)
+                # Not a label or a prefix.
+                else:
+                    # Test if argument
+                    try:
+                        args.append(float(part))
+                    except ValueError:
+                        op = part
```

Comment:
This is *definitely* the wrong way to handle this, if you had something like `push pop`, it would get translated into `pop`, and `pop 1 push` would get translated into `push 1`, etc.

Add some tests for cases like this.